test/cyanide_test.exs

```diff
@@ -0,0 +1,487 @@
+#
+# This file is part of Cyanide.
+#
+# Copyright 2019 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule CyanideTest do
```

Comment:
General notes:
- Split the Encoding testcases from the Decoding ones
- Tests shouls have name that read as a phrase (e.g.: `test "encodes an empty map")
- Try to have a single assertion per test or the amount of assertions that makes sense given the test name